browser/themes/brave_theme_service_browsertest.cc

```diff
@@ -305,36 +305,63 @@ IN_PROC_BROWSER_TEST_F(BraveThemeServiceTest, MAYBE_DarkModeChangeByRegTest) {
+  // system notifications of the dark mode changes.
   dark_mode::SetBraveDarkModeType(
       dark_mode::BraveDarkModeType::BRAVE_DARK_MODE_TYPE_DEFAULT);
 
-  apps_use_light_theme = !initial_dark_mode ? 0 : 1;
-  hkcu_themes_regkey.WriteValue(L"AppsUseLightTheme", apps_use_light_theme);
-
-  TestNativeThemeObserver native_theme_observer_for_default;
-  EXPECT_CALL(
-      native_theme_observer_for_default,
-      OnNativeThemeUpdated(ui::NativeTheme::GetInstanceForNativeUi())).Times(1);
-  ui::NativeTheme::GetInstanceForNativeUi()->AddObserver(
-      &native_theme_observer_for_default);
-
-  // Toggle dark mode and |native_theme_observer_for_light| will not get
-  // notification for light type.
-  dark_mode::SetBraveDarkModeType(
-      dark_mode::BraveDarkModeType::BRAVE_DARK_MODE_TYPE_LIGHT);
-
-  TestNativeThemeObserver native_theme_observer_for_light;
-  EXPECT_CALL(
-      native_theme_observer_for_light,
-      OnNativeThemeUpdated(ui::NativeTheme::GetInstanceForNativeUi())).Times(0);
-  ui::NativeTheme::GetInstanceForNativeUi()->AddObserver(
-      &native_theme_observer_for_light);
+  {
+    // Set up theme observer and toggle the system dark mode by writing to the
+    // registry. We should get 2 notifications:
+    //    1 for dark mode change + 1 for reduced transparency.
+    // We get notifications for both because they are watching the same registry
+    // key.
+    TestNativeThemeObserver native_theme_observer_for_default;
+    ui::NativeTheme::GetInstanceForNativeUi()->AddObserver(
+        &native_theme_observer_for_default);
+
+    EXPECT_CALL(native_theme_observer_for_default,
+                OnNativeThemeUpdated(ui::NativeTheme::GetInstanceForNativeUi()))
+        .Times(2);
+    apps_use_light_theme = !initial_dark_mode ? 0 : 1;
+    hkcu_themes_regkey.WriteValue(L"AppsUseLightTheme", apps_use_light_theme);
+
+    // Timeout is used to let notifications trickle in.
+    RunLoopRunWithTimeout(base::Milliseconds(500));
+
+    // Toggling dark mode to light should result in only one notification since
+    // we aren't touching the registry.
+    EXPECT_CALL(native_theme_observer_for_default,
+                OnNativeThemeUpdated(ui::NativeTheme::GetInstanceForNativeUi()))
+        .Times(1);
+    dark_mode::SetBraveDarkModeType(
+        dark_mode::BraveDarkModeType::BRAVE_DARK_MODE_TYPE_LIGHT);
 
-  apps_use_light_theme = initial_dark_mode ? 0 : 1;
-  hkcu_themes_regkey.WriteValue(L"AppsUseLightTheme", apps_use_light_theme);
+    ui::NativeTheme::GetInstanceForNativeUi()->RemoveObserver(
+        &native_theme_observer_for_default);
+  }
 
-  // Timeout is used because we can't get notifiication with light theme.
-  RunLoopRunWithTimeout(base::Milliseconds(500));
+  {
+    // Set up theme observer and toggle the system dark mode via the registry
+    // again. We should only get 1 reduced transparency notifications this time
+    // because we short circuit dark mode change system notifications when we
```

Comment:
This short circuit is upstream one?

Reply:
It's our patch that stops the notification propagation when we are not in the "Same as Windows" mode.